services/registration/src/routes/grading.ts

```diff
@@ -274,19 +292,20 @@ gradingRouter.route("/actions/submit-review").post(
         essayId: req.body.essayId,
         score: req.body.score,
         timestamp: new Date(),
-        // adjustedScore,
+        adjustedScore,
+      });
+
+      const allEssayReviews = await ReviewModel.find({
+        applicationId: { $eq: req.body.applicationId },
       });
 
-      // TODO: Needs to be fixed up, as all the essays need to be scored first
-      // const allEssayReviews = await ReviewModel.find({
-      //   essayId: req.body.essayId,
-      // });
-      // if (allEssayReviews.length >= MAX_REVIEWS_PER_ESSAY) {
-      //   application.gradingComplete = true;
-      //   const sumScores = allEssayReviews.reduce((prev, review) => prev + review.adjustedScore, 0);
-      //   application.finalScore = sumScores / allEssayReviews.length;
-      //   await application.save();
-      // }
+      // If all essays have been graded, compute the final score and mark the application as graded
+      if (allEssayReviews.length >= MAX_REVIEWS_PER_ESSAY * ESSAY_COUNT) {
+        application.gradingComplete = true;
+        const sumScores = allEssayReviews.reduce((prev, review) => prev + review.adjustedScore, 0);
+        application.finalScore = sumScores / allEssayReviews.length;
+        await application.save();
```

Comment:
If we're gonna add these fields to the application, we need to make sure on application GET requests, that if the user doesn't have member permission, this info is removed from their application. Could you update the routes for this? Want to make sure ppl can't see their scores.